enumerable.rb

```diff
@@ -0,0 +1,117 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity, Style/IdenticalConditionalBranches
+module Enumerable
+  def my_each
+    each do |i|
+      return to_enum(:my_each) unless block_given?
+
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    index = 0
+    each do |i|
+      return to_enum(:my_each_with_index) unless block_given?
+
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each { |item| new_arr << item if yield item }
+    new_arr
+  end
+
+  def my_all?(parameter = nil)
+    if block_given?
+      to_a.my_each { |items| return false if yield(items) == false }
+      return true
+    elsif parameter.nil?
+      to_a.my_each { |items| return false if items == false || items.nil? }
+    elsif !parameter.nil? && (parameter.is_a? Class)
+      to_a.my_each { |item| return false unless [item.class, item.class.superclass].include?(parameter) }
+    elsif !parameter.nil? && parameter.class == Regexp
+      to_a.my_each { |item| return false unless parameter.match(item) }
+    else
+      to_a.my_each { |item| return false if item != parameter }
+    end
+    true
+  end
+
+  def my_any?(parameter = nil)
+    if block_given?
+      to_a.my_each { |items| return true if yield(items) }
+      return false
+    elsif parameter.nil?
+      to_a.my_each { |items| return true if items }
+    elsif !parameter.nil? && (parameter.is_a? Class)
+      to_a.my_each { |items| return true if [items.class].include?(parameter) }
+    elsif !parameter.nil? && parameter.class == Regexp
+      to_a.my_each { |items| return true if parameter.match(items) }
+    else
+      to_a.my_each { |items| return true if items == parameter }
+    end
+    false
+  end
```

Comment:
When you pass a class say an array as an argument and there contains an array in the collection it should return true. Kindly recheck, there seems to be a problem. Test this method with a collection that contains an array. It seems to be returning false instead of true.[any](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F)

Reply:
We checked passing the class Array as an argument and applying the method to an array containing and array and it worked